spring-web/src/main/java/org/springframework/web/client/RestTemplate.java

```diff
@@ -684,8 +685,16 @@ public void delete(URI url) throws RestClientException {
 			response = request.execute();
 			handleResponse(url, method, response);
 			if (responseExtractor != null) {
-				return responseExtractor.extractData(response);
-			}
+                T entityData = responseExtractor.extractData(response);
+                if(entityData instanceof ResponseEntity){
+                    ResponseEntity responseEntity = (ResponseEntity)entityData;
+                    Object o = responseEntity.getBody();
+                    if(o instanceof InputStreamResource){
```

Comment:
I think better to use `InputStreamSource` instead of `InputStreamResource`.

Reply:
org.springframework.web.client.RestClientException: Could not extract response: no suitable HttpMessageConverter found for response type [interface org.springframework.core.io.InputStreamSource] and content type [application/octet-stream]
I need custom HttpMessageConverter ?